enum.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    range = is_a?(Range) ? to_a : self
+    par = is_a?(Range) ? size : length
+    while i < par
+      if is_a?(Hash)
+        yield(keys[i], self[keys[i]])
+      else
+        yield(range[i])
+      end
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    my_each do
+      if is_a?(Hash)
+        yield(self[keys[i]], keys[i])
+      else
+        yield(self[i], i)
+      end
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    my_each { |i| arr << i if yield(i) }
+    arr
+  end
+
+  def my_all?
+    my_each do |i|
+      if !arg[0].nil?
+        return false unless arg[0].equal? i
+      elsif block_given?
+        return false unless yield(i)
+      else
+        return false unless i
+      end
+    end
+    true
+  end
+
+  def my_any?(*arg)
+    my_each do |i|
+      if !arg[0].nil?
+        return true if arg[0].equal? i
+      elsif block_given?
+        return true if yield(i)
+      elsif i
+        return true
+      end
+    end
+    false
+  end
+
+  def my_none?(arg = nil, &block)
+    !my_any?(arg, &block)
+  end
```

Comment:
#### my_none?

- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class

     > this [1, 2.5, 'a', 9].my_none?(Integer) should be equal to this [1, 2.5, 'a', 9].none?(Integer) => false

- Yours is giving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/76905797-4c0c8d80-6868-11ea-9f3a-f4bbe4fbdfb8.png)

- Real
![image](https://user-images.githubusercontent.com/52209067/76905820-59297c80-6868-11ea-992c-3264fd368649.png)

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex

     > %w[dog door rod blade].my_none?(/o/) should be equal to this %w[dog door rod blade].none?(/o/) => false

- Yours is retrieving the wrong answer
![image](https://user-images.githubusercontent.com/52209067/76905915-8e35cf00-6868-11ea-9a23-085357c387ac.png)

- Real
![image](https://user-images.githubusercontent.com/52209067/76905954-9db51800-6868-11ea-8b05-bb9675e2ff16.png)